src/main/java/mil/dds/anet/WaitForDBCommand.java

```diff
@@ -0,0 +1,76 @@
+package mil.dds.anet;
+
+import org.skife.jdbi.v2.DBI;
+
+import io.dropwizard.cli.ConfiguredCommand;
+import net.sourceforge.argparse4j.inf.Namespace;
+import mil.dds.anet.config.AnetConfiguration;
+import io.dropwizard.Application;
+import io.dropwizard.setup.Bootstrap;
+import io.dropwizard.jdbi.DBIFactory;
+import io.dropwizard.setup.Environment;
+import net.sourceforge.argparse4j.inf.Subparser;
+
+
+public class WaitForDBCommand extends ConfiguredCommand<AnetConfiguration> {
+
+	public WaitForDBCommand() {
+		super("waitForDB","Waits until DB is ready for connection");
+	}
+
+	@Override
+	public void configure(Subparser subparser) {
+		subparser.addArgument("-n", "--nbAttempts")
+		.dest("dbConnectionNbAttempts")
+		.type(Integer.class)
+		.required(false)
+		.setDefault(20)
+		.help("Nb of attempts before giving up. 20 by default");
+
+		subparser.addArgument("-d", "--delay")
+		.dest("dbConnectionDelay")
+		.type(Integer.class)
+		.required(false)
+		.setDefault(500)
+		.help("Delay in ms between attempts. 500 by default");
+
+		addFileArgument(subparser);
+	}
+
+	@Override
+	protected void run(Bootstrap<AnetConfiguration> bootstrap, Namespace namespace, AnetConfiguration configuration)
+		throws Exception {
+			final DBIFactory factory = new DBIFactory();
+			final Environment environment = new Environment(bootstrap.getApplication().getName(),
+					bootstrap.getObjectMapper(),
+					bootstrap.getValidatorFactory().getValidator(),
+					bootstrap.getMetricRegistry(),
+					bootstrap.getClassLoader(),
+					bootstrap.getHealthCheckRegistry());
+			final DBI jdbi = factory.build(environment, configuration.getDataSourceFactory(), "mssql");
+
+		//We want to possibly wait for the database to be ready, and keep trying to connect
+		int remainingTries = namespace.getInt("dbConnectionNbAttempts").intValue();
+		final int delay = namespace.getInt("dbConnectionDelay").intValue();
+		Exception dbiException;
+		do {
+			try {
+				jdbi.close(jdbi.open());
+				dbiException = null;
+				break;
+			}
+			catch (Exception exception){
+				dbiException = exception;
+			}
+			try{
+				Thread.sleep(delay);
```

Comment:
This sleep should be done only when `remainingTries > 0` else you wait unnecessarily after the final failed attempt.